tests/tensor_search/test_model_auth.py

```diff
@@ -991,6 +1011,1372 @@ def test_open_clip_reg_clip(self):
+        )
+
+        # is the hf model being loaded with the expected args?
+        assert len(mock_automodel_from_pretrained.call_args_list) == 1
+        assert mock_automodel_from_pretrained.call_args_list[0][0][0] == 'cache/path/to/model/', "Expected call not found"
+
+        # is the hf tokenizer being loaded with the expected args?
+        assert len(mock_autotokenizer_from_pretrained.call_args_list) == 1
+        assert mock_autotokenizer_from_pretrained.call_args_list[0][0][0] == 'cache/path/to/model/', "Expected call not found"
+
+        # is the zip file being extracted with the expected args?
+        assert len(mock_validate_huggingface_archive.call_args_list) == 1
+        assert mock_validate_huggingface_archive.call_args_list[0][0][0] == 'cache/path/to/model.zip', "Expected call not found"
+
+    def test_3_load_model_from_s3_zip_file_with_auth_search(self):
+        s3_object_key = 'path/to/your/secret_model.pt'
+        s3_bucket = 'your-bucket-name'
+
+        _delete_file(os.path.join(ModelCache.hf_cache_path, os.path.basename(s3_object_key)))
+
+        model_properties = {
+            "dimensions": 384,
+            "model_location": {
+                "s3": {
+                    "Bucket": s3_bucket,
+                    "Key": s3_object_key,
+                },
+                "auth_required": True
+            },
+            "type": "hf",
+        }
+        s3_settings = _get_base_index_settings()
+        s3_settings['index_defaults']['model_properties'] = model_properties
+        tensor_search.create_vector_index(config=self.config, index_name=self.index_name_1, index_settings=s3_settings)
+
+        fake_access_key_id = '12345'
+        fake_secret_key = 'this-is-a-secret'
+        public_model_url = "https://dummy/url/for/model.zip"
+
+        # Create a mock Boto3 client
+        mock_s3_client = mock.MagicMock()
+
+        # Mock the generate_presigned_url method of the mock Boto3 client to return a dummy URL
+        mock_s3_client.generate_presigned_url.return_value = public_model_url
+
+        with unittest.mock.patch('boto3.client', return_value=mock_s3_client) as mock_boto3_client:
+
+            try:
+                res = tensor_search.search(
+                    config=self.config, text='hello', index_name=self.index_name_1,
+                    model_auth=ModelAuth(s3=S3Auth(aws_access_key_id=fake_access_key_id, aws_secret_access_key=fake_secret_key))
+                )
+            except BadRequestError as e:
+                print()
+                pass
+
+        mock_s3_client.generate_presigned_url.assert_called_with(
+            'get_object',
+            Params={'Bucket': 'your-bucket-name', 'Key': s3_object_key}
+        )
+        mock_boto3_client.assert_called_once_with(
+            's3',
+            aws_access_key_id=fake_access_key_id,
+            aws_secret_access_key=fake_secret_key,
+            aws_session_token=None
+        )
```

Comment:
is it worth asserting `from_pretrained` is called with correct args?

Reply:
updated with extra mock test